### ISSUE TITLE
Regarding the Lovasz Hinge loss

### ISSUE THREAD

**Issue by MLbyML** [davyneven/SpatialEmbeddings#14]
Hello,

I have more of a question here rather than an issue. I was curious about this [line](https://github.com/davyneven/SpatialEmbeddings/blob/213b5fa9900513818365d98e7db9f2067633d94d/src/criterions/my_loss.py#L100) of code. As I understand, `dist` is a variable that lies between 0 (pixel embedding is very far from the instance center) and 1 (pixel embedding lies atop the instance center). Also by that logic, 

```python3
0< dist*2 -1 <1
```
Could you share any intuition on why this `dist*2-1` is preferred instead of just using `dist`, as the first argument to the `lovasz hinge` class? Does it have better convergence properties for the loss in your opinion, for example? Thank you!

**Comment by JaWeyl** [comment]
Hi,

I just wanted to share my thoughts with you since I stumbled about the same problem.

The signature of the loss function is `lovasz_hinge(logits, labels, per_image=True, ignore=None) ` - so it expects logits in the range of [-\infty. +\infty] instead of probabilities [0, 1]. 

By computing `dist*2 -1` the author maps all predicted probabilities from [0,1] to [-1,1] to transform them to valid logits.
So from my point of view this mapping is necessary due to the signature of the loss function.

**Comment by davyneven** [comment]
Indeed, I map the distance/probability to the interval [-1,1], as expected by the lovasz loss.

**Comment by MLbyML** [comment]
Thanks for the reply @JaWeyl and @davyneven That was helpful!
